tonic/src/transport/server/mod.rs

```diff
@@ -333,7 +333,7 @@ impl Server {
         };
 
         let server = hyper::Server::builder(incoming)
-            .http2_only(true)
+            .http2_only(false)
```

Comment:
This is likely to cause issues if we turn this off by default. We should think about this impact.

Reply:
I think it should be on by default. To me, the ideal api to wire up gpr-web is very close to what was proposed in #288, something like:

```rust
Server::builder
    .add_service(..)
    .add_service(..)
    .accept_grpc_web(Cors::default())
    .server(addr);
```

where the builder arranges everything internally and either all services attached to the server are capable of responding to grpc-web requests or none are. This prevents misconfigurations and correct but perplexing protocol or decoding errors that an api like this will cause:

```rust
Server::builder
    .add_service(GrpcWeb::new(..))
    .add_service(..)
    .enable_http1(..)
    .server(addr);
```

where it is possible to:

* Enable http1 unnecessarily 
* Wrap services in GrpcWeb and forget to enable http1
* Enabe http1 and forget to wrap some or all services in GrpcWeb

Reply:
The GrpcWeb service, however it ends up being wired-up, would need to handle at least one more case when enabled: http1 requests with `application/grpc` content type. In this case it should probably just return bad request.

Reply:
One more wrinkle that would probably be nice to handle: Enable serving gprc-web requests but *not* http1, but not the other way around. 

I will address your other comments in an upcoming commit and will explore this further once the core translation is ready.

Reply:
yeah, we will need to also ensure alpn still works and maybe also support h2c style upgrades.

Reply:
I'll have to confirm but I _think_ alpn works. I don't think h2c is necessary because browsers do not support it anyway.